src/udev/udevadm-info.c

```diff
@@ -1054,149 +1219,134 @@ int info_main(int argc, char *argv[], void *userdata) {
-                                return r;
-
-                        r = sd_device_enumerator_add_match_sysname(e, optarg);
+                        r = strv_extend(&arg_sysname_match, optarg);
                         if (r < 0)
-                                return log_error_errno(r, "Failed to add sysname match '%s': %m", optarg);
+                                return log_oom();
                         break;
 
                 case ARG_NAME_MATCH:
-                case ARG_PARENT_MATCH: {
-                        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
-
-                        r = find_device(optarg, c == ARG_NAME_MATCH ? "/dev" : "/sys", &dev);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to open the device '%s': %m", optarg);
-
-                        r = ensure_device_enumerator(&e);
+                        r = strv_extend(&arg_name_match, optarg);
                         if (r < 0)
-                                return r;
+                                return log_oom();
+                        break;
 
-                        r = device_enumerator_add_match_parent_incremental(e, dev);
+                case ARG_PARENT_MATCH:
+                        r = strv_extend(&arg_parent_match, optarg);
                         if (r < 0)
-                                return log_error_errno(r, "Failed to add parent match '%s': %m", optarg);
+                                return log_oom();
                         break;
-                }
 
                 case ARG_INITIALIZED_MATCH:
-                case ARG_INITIALIZED_NOMATCH:
-                        r = ensure_device_enumerator(&e);
-                        if (r < 0)
-                                return r;
+                        arg_initialized_match = MATCH_INITIALIZED_YES;
+                        break;
 
-                        r = device_enumerator_add_match_is_initialized(e, c == ARG_INITIALIZED_MATCH ? MATCH_INITIALIZED_YES : MATCH_INITIALIZED_NO);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to set initialized filter: %m");
+                case ARG_INITIALIZED_NOMATCH:
+                        arg_initialized_match = MATCH_INITIALIZED_NO;
                         break;
 
                 case '?':
                         return -EINVAL;
+
                 default:
                         assert_not_reached();
                 }
 
-        if (action == ACTION_DEVICE_ID_FILE) {
-                if (argv[optind])
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                               "Positional arguments are not allowed with -d/--device-id-of-file.");
-                assert(name);
-                return stat_device(name, arg_export, arg_export_prefix);
-        }
+        r = strv_extend_strv(&arg_devices, argv + optind, /* filter_duplicates= */ false);
```

Comment:
strv_skip() ?

Reply:
Is it necessary? We know that `optind < argc` at this point (unless there's something seriously wrong with `getopt_long()`).

Reply:
Not necessary. It is just my preference. Feel free to ignore.